Type definition changes in 1.3.12 break usage with `extends`

We have our config defined as follows:

```typescript
import vitestPlugin from "@vitest/eslint-plugin";
import { defineConfig } from "eslint/config";

export const vitestConfig = defineConfig({
  files: ["**/*.test.ts"],
  extends: [vitestPlugin.configs.recommended],
  settings: {
    vitest: {
      typecheck: true,
    },
  }
});
```

This worked in v1.3.10 but breaks in 1.3.12:

```
eslint-plugin-vitest.ts:6:13 - error TS2322: Type '{ readonly name: "vitest/recommended"; readonly plugins: { readonly vitest: { readonly meta: { readonly name: "vitest"; readonly version: string; }; readonly rules: { readonly "prefer-lowercase-title": RuleModule<MessageIds, [...], PluginDocs, RuleListener>; ... 71 more ...; readonly "warn-todo": RuleModule<...>; };...' is not assignable to type 'ExtendsElement'.
  Type '{ readonly name: "vitest/recommended"; readonly plugins: { readonly vitest: { readonly meta: { readonly name: "vitest"; readonly version: string; }; readonly rules: { readonly "prefer-lowercase-title": RuleModule<MessageIds, [...], PluginDocs, RuleListener>; ... 71 more ...; readonly "warn-todo": RuleModule<...>; };...' is not assignable to type 'Config<RulesRecord>'.
    Types of property 'plugins' are incompatible.
      Type '{ readonly vitest: { readonly meta: { readonly name: "vitest"; readonly version: string; }; readonly rules: { readonly "prefer-lowercase-title": RuleModule<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>], PluginDocs, RuleL...' is not assignable to type 'Record<string, Plugin>'.
        Property 'vitest' is incompatible with index signature.
          Type '{ readonly meta: { readonly name: "vitest"; readonly version: string; }; readonly rules: { readonly "prefer-lowercase-title": RuleModule<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>], PluginDocs, RuleListener>; ... 71 mo...' is not assignable to type 'Plugin'.
            Types of property 'rules' are incompatible.
              Type '{ readonly "prefer-lowercase-title": RuleModule<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>], PluginDocs, RuleListener>; ... 71 more ...; readonly "warn-todo": RuleModule<...>; }' is not assignable to type 'Record<string, RuleDefinition<RuleDefinitionTypeOptions>>'.
                Property '"prefer-lowercase-title"' is incompatible with index signature.
                  Type 'RuleModule<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>], PluginDocs, RuleListener>' is not assignable to type 'RuleDefinition<RuleDefinitionTypeOptions>'.
                    Types of property 'create' are incompatible.
                      Type '(context: Readonly<RuleContext<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>]>>) => RuleListener' is not assignable to type '(context: RuleContext<{ LangOptions: LanguageOptions; Code: SourceCode<{ LangOptions: LanguageOptions; RootNode: unknown; SyntaxElementWithLoc: unknown; ConfigNode: unknown; }>; RuleOptions: unknown[]; Node: unknown; MessageIds: string; }>) => RuleVisitor'.
                        Types of parameters 'context' and 'context' are incompatible.
                          Type 'RuleContext<{ LangOptions: LanguageOptions; Code: SourceCode<{ LangOptions: LanguageOptions; RootNode: unknown; SyntaxElementWithLoc: unknown; ConfigNode: unknown; }>; RuleOptions: unknown[]; Node: unknown; MessageIds: string; }>' is missing the following properties from type 'Readonly<RuleContext<MessageIds, [Partial<{ ignore: string[]; allowedPrefixes: string[]; ignoreTopLevelDescribe: boolean; lowercaseFirstCharacterOnly: boolean; }>]>>': getAncestors, getDeclaredVariables, getScope, markVariableAsUsed

6   extends: [vitestPlugin.configs.recommended],
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error.
```

I believe the addition of `readonly` to all the properties has caused this incompatibility?

Repro: https://github.com/haines/eslint-plugin-vitest-types

--- a/README.md
+++ b/README.md
@@ -24,20 +24,19 @@ npm install @vitest/eslint-plugin --save-dev
 Make sure you're running ESLint `v9.0.0` or higher for the latest version of this plugin to work. The following example is how your `eslint.config.js` should be setup for this plugin to work for you.
 
 ```js
+import { defineConfig } from 'eslint/config'
 import vitest from '@vitest/eslint-plugin'
 
-export default [
-  {
-    files: ['tests/**'], // or any other pattern
-    plugins: {
-      vitest,
-    },
-    rules: {
-      ...vitest.configs.recommended.rules, // you can also use vitest.configs.all.rules to enable all rules
-      'vitest/max-nested-describe': ['error', { max: 3 }], // you can also modify rules' behavior using option like this
-    },
+export default defineConfig({
+  files: ['tests/**'], // or any other pattern
+  plugins: {
+    vitest,
   },
-]
+  rules: {
+    ...vitest.configs.recommended.rules, // you can also use vitest.configs.all.rules to enable all rules
+    'vitest/max-nested-describe': ['error', { max: 3 }], // you can also modify rules' behavior using option like this
+  },
+})
 ```
 
 If you're not using the latest version of ESLint (version `v8.57.0` or lower) you can setup this plugin using the following configuration
@@ -80,9 +79,20 @@ Vitest ships with an optional [type-testing feature](https://vitest.dev/guide/te
 If you're using this feature, you should also enabled `typecheck` in the settings for this plugin. This ensures that rules like [expect-expect](docs/rules/expect-expect.md) account for type-related assertions in tests.
 
 ```js
+import { defineConfig } from 'eslint/config'
+import tseslint from 'typescript-eslint'
 import vitest from '@vitest/eslint-plugin'
 
-export default [
+export default defineConfig(
+  // see https://typescript-eslint.io
+  tseslint.configs.recommended,
+  {
+    languageOptions: {
+      parserOptions: {
+        projectService: true,
+      },
+    },
+  },
   {
     files: ['tests/**'], // or any other pattern
     plugins: {
@@ -102,7 +112,7 @@ export default [
       },
     },
   },
-]
+)
 ```
 
 ### Custom Fixtures
@@ -110,24 +120,23 @@ export default [
 If you're using custom fixtures in a separate file and importing them in your tests, you can let the plugin know about them by adding them to the `vitestImports` setting. The property accepts an array of strings or regular expressions that match the module names where your custom fixtures are defined.
 
 ```js
+import { defineConfig } from 'eslint/config'
 import vitest from '@vitest/eslint-plugin'
 
-export default [
-  {
-    files: ['tests/**'], // or any other pattern
-    plugins: {
-      vitest,
-    },
-    rules: {
-      ...vitest.configs.recommended.rules,
-    },
-    settings: {
-      vitest: {
-        vitestImports: ['@/tests/fixtures', /test-extend$/],
-      },
+export default defineConfig({
+  files: ['tests/**'], // or any other pattern
+  plugins: {
+    vitest,
+  },
+  rules: {
+    ...vitest.configs.recommended.rules,
+  },
+  settings: {
+    vitest: {
+      vitestImports: ['@/tests/fixtures', /test-extend$/],
     },
   },
-]
+})
 ```
 
 ### Shareable Configurations
@@ -139,14 +148,13 @@ This plugin exports a recommended configuration that enforces good testing pract
 To enable this configuration with `eslint.config.js`, use `vitest.configs.recommended`:
 
 ```js
+import { defineConfig } from 'eslint/config'
 import vitest from '@vitest/eslint-plugin'
 
-export default [
-  {
-    files: ['tests/**'], // or any other pattern
-    ...vitest.configs.recommended,
-  },
-]
+export default defineConfig({
+  files: ['tests/**'], // or any other pattern
+  ...vitest.configs.recommended,
+})
 ```
 
 #### All
@@ -154,14 +162,13 @@ export default [
 If you want to enable all rules instead of only some you can do so by adding the all configuration to your `eslint.config.js` config file:
 
 ```js
+import { defineConfig } from 'eslint/config'
 import vitest from '@vitest/eslint-plugin'
 
-export default [
-  {
-    files: ['tests/**'], // or any other pattern
-    ...vitest.configs.all,
-  },
-]
+export default defineConfig({
+  files: ['tests/**'], // or any other pattern
+  ...vitest.configs.all,
+})
 ```
 
 ### Rules

--- a/eslint.config.ts
+++ b/eslint.config.ts
@@ -10,7 +10,6 @@ export default defineConfig(
   gitignore(),
   eslint.configs.recommended,
   tseslint.configs.recommended,
-  // @ts-expect-error see https://github.com/vitest-dev/eslint-plugin-vitest/issues/771
   vitest.configs.recommended,
   eslintPlugin.configs.recommended,
   {

--- a/src/index.ts
+++ b/src/index.ts
@@ -1,4 +1,4 @@
-import type { Linter } from 'eslint'
+import type { ESLint, Linter, Rule } from 'eslint'
 import { version } from '../package.json'
 import lowerCaseTitle, { RULE_NAME as lowerCaseTitleName } from './rules/prefer-lowercase-title'
 import maxNestedDescribe, { RULE_NAME as maxNestedDescribeName } from './rules/max-nested-describe'
@@ -262,7 +262,7 @@ const rules = {
     [warnTodoName]: warnTodo,
     [preferImportInMockName]: preferImportInMock,
     [preferCalledExactlyOnceWithName]: preferCalledExactlyOnceWith
-} as const
+} as unknown as Record<string, Rule.RuleModule>
 
 const plugin = {
   meta: {
@@ -299,7 +299,7 @@ const plugin = {
     recommended: {
       name: 'vitest/recommended',
       plugins: {
-        get vitest() {
+        get vitest(): ESLint.Plugin {
           return plugin
         },
       },
@@ -308,7 +308,7 @@ const plugin = {
     all: {
       name: 'vitest/all',
       plugins: {
-        get vitest() {
+        get vitest(): ESLint.Plugin {
           return plugin
         },
       },
@@ -337,8 +337,8 @@ const plugin = {
           onTestFinished: 'writable',
         },
       },
-    } as const satisfies Linter.Config,
-  } as const,
-} as const
+    },
+  },
+} as const satisfies ESLint.Plugin
 
 export default plugin

